### PR TITLE
Implement support for `[LegacyWindowAlias]` and `[NoInterfaceObject]`

### LINKED ISSUE #187: [LegacyWindowAlias]
Used by [DOMRect](https://drafts.fxtf.org/geometry/#domrect) and friends.

### PR DESCRIPTION
**Pull request by ExE-Boss**
Fixes #187

---

review?(@TimothyGu)

### CODE REVIEW

File: test/cases/DOMRect.webidl [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+interface DOMRect /* : DOMRectReadOnly */ {
+  constructor(optional unrestricted double x = 0, optional unrestricted double y = 0,
+          optional unrestricted double width = 0, optional unrestricted double height = 0);
+
+  [NewObject] static DOMRect fromRect(optional DOMRectInit other);
+
+  inherit attribute unrestricted double x;
+  inherit attribute unrestricted double y;
+  inherit attribute unrestricted double width;
+  inherit attribute unrestricted double height;
```
Comment (TimothyGu) [inline test/cases/DOMRect.webidl:15]: Technically `inherit` should not be there for interfaces that don't inherit another interface.

```suggestion
  attribute unrestricted double x;
  attribute unrestricted double y;
  attribute unrestricted double width;
  attribute unrestricted double height;
```

File: lib/constructs/interface.js [hunk 2]
```diff
@@ -1444,12 +1487,39 @@ class Interface {
...
         Object.defineProperty(globalObject, interfaceName, {
           configurable: true,
           writable: true,
           value: ${name}
         });
+      `;
+
+      if (this.legacyWindowAliases) {
+        this.str += `
+        if (globalName === "Window") {
```
Comment (TimothyGu) [inline lib/constructs/interface.js:1503]:
```suggestion
          if (globalName === "Window") {
```

The existing convention is to indent the template literal by at least one tab level.

File: lib/constructs/interface.js [hunk 3]
```diff
@@ -1444,12 +1487,39 @@ class Interface {
...
+        `;
+
+        for (const legacyWindowAlias of this.legacyWindowAliases) {
+          this.str += `
+          Object.defineProperty(globalObject, "${legacyWindowAlias}", {
+            configurable: true,
+            writable: true,
+            value: ${name}
+          });
```
Comment (TimothyGu) [inline lib/constructs/interface.js:1512]:
```suggestion
            Object.defineProperty(globalObject, "${legacyWindowAlias}", {
              configurable: true,
              writable: true,
              value: ${name}
            });
```

File: lib/constructs/interface.js [hunk 4]
```diff
@@ -1444,12 +1487,39 @@ class Interface {
...
+          `;
+        }
+
+        this.str += `
+        }
```
Comment (TimothyGu) [inline lib/constructs/interface.js:1517]:
```suggestion
          }
```

File: lib/constructs/interface.js (hunk 2 above)
Comment (ExE-Boss) [reply lib/constructs/interface.js]: I matched the indentation of the other template literals.

**Comment by ExE-Boss** [comment]
`[Exposed]` is being implemented in <https://github.com/jsdom/webidl2js/pull/190>.

File: lib/constructs/interface.js [hunk 5]
```diff
@@ -1420,7 +1463,7 @@ class Interface {
     const { idl, name } = this;
 
     this.str += `
-      exports.install = function install(globalObject) {
+      exports.install = (globalObject, globalName) => {
```
Comment (ExE-Boss) [inline lib/constructs/interface.js:1466]: After re‑reading the spec for `[Global]` and `[Exposed]`, I realised that this needs `globalNames` instead of `globalName`.

**Pull request merged**